src/infoPanel/infoPanel.c

```diff
@@ -112,55 +147,95 @@ int main(int argc, char *argv[])
 				strncpy(image_path, argv[++i], STR_MAX-1);
 			else if (strcmp(argv[i], "-j") == 0 || strcmp(argv[i], "--images-json") == 0)
 				strncpy(images_json_path, argv[++i], STR_MAX-1);
+			else if (strcmp(argv[i], "-d") == 0 || strcmp(argv[i], "--directory") == 0)
+				strncpy(images_dir_path, argv[++i], STR_MAX-1);
+			else if (strcmp(argv[i], "-s") == 0 || strcmp(argv[i], "--show-theme-controls") == 0)
+				show_theme_controls = true;
 			else if (strcmp(argv[i], "-a") == 0 || strcmp(argv[i], "--auto") == 0)
 				wait_confirm = false;
 		}
 	}
 
-	SDL_Init(SDL_INIT_VIDEO);
-	SDL_ShowCursor(SDL_DISABLE);
-	TTF_Init();
-	
-	SDL_Surface *video = SDL_SetVideoMode(640, 480, 32, SDL_HWSURFACE);
-	SDL_Surface *screen = SDL_CreateRGBSurface(SDL_HWSURFACE, 640, 480, 32, 0, 0, 0, 0);
+	signal(SIGINT, sigHandler);
+    signal(SIGTERM, sigHandler);
+
+	SDL_InitDefault(true);
+
+	settings_load();
+	lang_load();
+
+	int battery_percentage = battery_getPercentage();
+
+	uint32_t acc_ticks = 0;
+	uint32_t last_ticks = SDL_GetTicks();
+	uint32_t time_step = 1000 / FRAMES_PER_SECOND;
 
 	bool cache_used = false;
-	if (exists(image_path)) {
+	if (exists(image_path))
+	{
 		g_images_paths_count = 1;
+		g_image_index = 0;
+		SDL_BlitSurface(theme_background(), NULL, screen, NULL);
 		drawImage(image_path, screen);
 	}
 	else if(exists(images_json_path))
 	{
-		if (loadImagesPaths(images_json_path, &g_images_paths, &g_images_paths_count))
+		if (loadImagesPathsFromJson(images_json_path, &g_images_paths, &g_images_paths_count, &g_images_titles)
+			&& g_images_paths_count > 0)
 		{
-			if (g_images_paths_count > 0)
-			{
-				drawImageByIndex(0, g_image_index, g_images_paths, g_images_paths_count, screen, &cache_used);
-			}
+			g_image_index = 0;
+			SDL_BlitSurface(theme_background(), NULL, screen, NULL);
+			drawImageByIndex(0, g_image_index, g_images_paths, g_images_paths_count, screen, &cache_used);
 		}
 		else
 		{
 			sdlQuit(screen, video);
 			return EXIT_FAILURE;
 		}
 	}
-	else {
-		drawInfoPanel(screen, video, title_str, message_str);
+	else if (exists(images_dir_path))
+	{
+		if (loadImagesPathsFromDir(images_dir_path, &g_images_paths, &g_images_paths_count)
+			&& g_images_paths_count > 0)
+		{
+			g_image_index = 0;
+			SDL_BlitSurface(theme_background(), NULL, screen, NULL);
+			drawImageByIndex(0, g_image_index, g_images_paths, g_images_paths_count, screen, &cache_used);
+		}
+		else
+		{
+			sdlQuit(screen, video);
+			return EXIT_FAILURE;
+		}
+	}
+	else if (strlen(title_str) > 0 && strlen(message_str) > 0)
+	{
+		info_panel_mode = true;
+		show_theme_controls = true;
+	}
+	else
+	{
+		sdlQuit(screen, video);
+		return EXIT_FAILURE;
 	}
 
-	SDL_BlitSurface(screen, NULL, video, NULL);
-	SDL_Flip(video);
-	
 	bool quit = false;
```

Comment:
This line needs to be deleted, you are using the global variable `quit` (else, the sigHandler won't work).